resources/assets/js/videos/videoContainer.vue

```diff
@@ -425,7 +425,9 @@ export default {
             this.attachAnnotationLabel(annotation)
                 .then(() => {
                     if (lastLabel) {
-                        this.detachAnnotationLabel(annotation, lastLabel);
+                        this.detachAnnotationLabel(annotation, lastLabel)
+                            .then(Events.$emit('video.swap', annotation));
```

Comment:
As an aside: Line 429 would not work correctly because the event would be dispatched immediately and not after the promise is resolved. This would be correct:

```suggestion
                            .then(() => Events.$emit('video.swap', annotation));
```